BazelExtensions/xcode_configuration_provider.bzl

```diff
@@ -239,6 +239,28 @@ def _objc_cmd_line(target, ctx, src, non_arc_srcs, output_file_path):
 
     return cmd_line_args
 
+def _swift_cmd_line(target, src):
+    argv = []
+    # In theory a single SwiftCompile action should represent
+    # the compilation of `src`, so fails if more than one is found
+    for a in target.actions:
+        if a.argv and a.mnemonic == "SwiftCompile" and src.path in a.argv:
+            argv.append(a.argv)
+
+    if len(argv) != 1:
+        fail("[ERROR] Source file {} referenced in multiple SwiftCompile actions.".format(src.path))
```

Comment:
Can you add this fail conditionally if it's a `SwiftCompile`? There's some other menemonics that will compile which are commonly used